### PR TITLE
Query: Re-organize SelectExpression

### PR DESCRIPTION
**Pull request by smitpatel**
Part of #17337

This commit doesn't make any functional change. Just moved code around for better organization and made certain private functions local static functions

### CODE REVIEW

File: src/EFCore.Relational/Query/SqlExpressions/SelectExpression.Helper.cs [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq.Expressions;
+using Microsoft.EntityFrameworkCore.Metadata;
+using Microsoft.EntityFrameworkCore.Utilities;
+
+namespace Microsoft.EntityFrameworkCore.Query.SqlExpressions
+{
+    public sealed partial class SelectExpression
```
Comment (roji) [inline src/EFCore.Relational/Query/SqlExpressions/SelectExpression.Helper.cs:13]: Maybe call the file SelectExpression.Visitors.cs?

**Review by roji — APPROVED** [review]
:sheep: :it:

File: src/EFCore.Relational/Query/SqlExpressions/SelectExpression.Helper.cs (hunk 1 above)
Comment (smitpatel) [reply src/EFCore.Relational/Query/SqlExpressions/SelectExpression.Helper.cs:13]: I will do rename if at the end of work, they all are visitors. I think I have a custom comparer nested class too to put here. Any other suggestions?

Comment (roji) [reply src/EFCore.Relational/Query/SqlExpressions/SelectExpression.Helper.cs:13]: Nope - LGTM. Didn't do a very thorough review, but as this is just code organization...

**Pull request merged**